### PR TITLE
feat: Add initial version of noir.js

### PR DESCRIPTION
**Pull request by jonybur**
# Description

A simple package that exports `acvm_js` and `noirc_abi`

## Problem\*

Resolves https://github.com/noir-lang/Planning/issues/32, https://github.com/noir-lang/Planning/issues/31

## Summary\*

Creates a package that exports acvm_js and noirc_abi

## Documentation

- [ ] This PR requires documentation updates when merged.

  

  - [ ] I will submit a noir-lang/docs PR.

  

  - [ ] I will request for and support Dev Rel's help in documenting this PR.

  
  

## Additional Context

# PR Checklist\*

- [ ] I have tested the changes locally.
- [ ] I have formatted the changes with [Prettier](https://prettier.io/) and/or `cargo fmt` on default settings.

### CODE REVIEW

File: tooling/noir_js/yarn.lock
Comment (TomAFrench) [inline tooling/noir_js/yarn.lock:1]: This shouldn't need to exist if the package is in the workspace.

Comment (kevaundray) [reply tooling/noir_js/yarn.lock:1]: removed

File: .github/workflows/noir-js.yml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+          fi
+          echo "NEW_VERSION=$NEW_VERSION" >> $GITHUB_ENV
+
+      - name: Install dependencies
+        working-directory: ./tooling/noir_js
+        run: npm install
+
+      - name: Build noir-source-resolver
+        working-directory: ./tooling/noir_js
+        run: npm run build
```
Comment (TomAFrench) [inline .github/workflows/noir-js.yml:35]:
```suggestion
      - name: Install dependencies
        run: yarn install --immutable

      - name: Build noir_js
        run: yarn workspace @noir-lang/noir_js build
```

We need to use `yarn` everywhere as `npm` won't respect our lockfiles.

File: .github/workflows/noir-js.yml [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+          NODE_AUTH_TOKEN: ${{ secrets.NPM_PUBLISH_TOKEN }}
+
+      - name: Configure git
+        run: |
+          git config user.name kevaundray
+          git config user.email kevtheappdev@gmail.com
+
+      - name: Commit updates
+        run: |
+          git add tooling/noir_js/package-lock.json
```
Comment (TomAFrench) [inline .github/workflows/noir-js.yml:50]:
```suggestion
          git add yarn.lock
```

We shouln't have `package-lock.json` ever.

**Review by TomAFrench — DISMISSED** [review]
LGTM

File: .github/workflows/noir-js.yml [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Bump version
+        working-directory: ./tooling/noir_js
+        id: bump_version
+        run: |
+          if [ -z "${{ github.event.inputs.version }}" ]; then
+            NEW_VERSION=$(npm version patch --no-git-tag-version)
```
[the remark this answers is not in the document]
Comment (TomAFrench) [reply .github/workflows/noir-js.yml:23]: https://github.com/noir-lang/noir/pull/2704

[the remark this answers is not in the document]
Comment (TomAFrench) [reply .github/workflows/noir-js.yml:23]: Agree on having this handled by release-please. We'll probably need to set up a few scopes in order to release things at the appropriate times.

**Comment by TomAFrench** [comment]
On second thoughts, might it be safer to require the version bump to have been done already manually?

This makes running the release workflow idempotent.

**Comment by TomAFrench** [comment]
Not a blocker.

**Comment by kevaundray** [comment]
> require the version bump to have been done already manually?

yeah I don't mind doing that

**Comment by TomAFrench** [comment]
Actually, let's merge this and then we can address both workflows together.

**Comment by kevaundray** [comment]
Oh sorry already pushed it

**Review by TomAFrench — APPROVED** [review]

**Pull request merged**